internal/http/routes.go

```diff
@@ -2,13 +2,26 @@ package http
 
 import (
 	"github.com/go-chi/chi/v5"
+
+	"github.com/Kenny201/go-yandex-shortener.git/internal/http/middleware"
 )
 
-// Routes returns the initialized router
 func (s *Server) useRoutes() *chi.Mux {
 	r := chi.NewRouter()
-	r.Post("/", s.shortenerHandler.PostHandler)
-	r.Get("/{id}", s.shortenerHandler.GetByIDHandler)
+
+	r.Use(
+		middleware.Gzip,
+		middleware.Logger,
+	)
+
+	r.Post("/", s.handler.Post)
+	r.Get("/{id}", s.handler.Get)
+
+	r.Route("/api", func(r chi.Router) {
+		r.Route("/shorten", func(r chi.Router) {
+			r.Post("/", s.handler.PostAPI)
+		})
+	})
```

Comment:
Кажется уже писал, что технически во сложениях вызовов нет смысла. Проще укоротить это до:
```suggestion
	r.Route("/api", func(r chi.Router) {
		r.Post("/shorten", s.handler.PostAPI)
	})
```